octofree/scraper_x.py

```
@@ -8,21 +23,69 @@
+        params (dict): Query parameters for the request.
+    
+    Returns:
+        dict or None: Parsed JSON response, or None if request fails.
+            Logs error details on failure.
+    """
+    try:
+        response = requests.request("GET", url, auth=bearer_oauth, params=params, timeout=30)
+        response.raise_for_status()  # Raise exception for non-2xx status codes
+        return response.json()
+    except requests.RequestException as e:
+        logging.error(f"[SCRAPER_X] Request to {url} failed: {e}")
```

Ruff (0.14.1)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)